tests/parser-test-files/define-class.txt

```diff
@@ -13,3 +336,154 @@ error: Malformed class definition
+
+(package test-package)
+
+(define-class (C :a)
+  (0.5 (:a -> :a)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed method definition
+  --> test:4:3
+   |
+ 3 |  (define-class (C :a)
+   |                ------ in this class definition
+ 4 |    (0.5 (:a -> :a)))
+   |     ^^^ expected symbol
+
+================================================================================
+113 Malformed class definition: dotten list
+================================================================================
+
+(package test-package)
+
+(define-class (C :a . :b))
+
+--------------------------------------------------------------------------------
+
+error: Malformed class definition
+  --> test:3:14
+   |
+ 3 |  (define-class (C :a . :b))
+   |                ^^^^^^^^^^^ unexpected dotted list
+
+================================================================================
+114 Malformed functional dependency: atom in fundeps
+================================================================================
+
+(package test-package)
+
+(define-class (C :a :b (:a -> :b) :c))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:34
+   |
+ 3 |  (define-class (C :a :b (:a -> :b) :c))
+   |                                    ^^ expected a list
+
+================================================================================
+115 Malformed functional dependency: dotted list
+================================================================================
+
+(package test-package)
+
+(define-class (C :a :b (:a :b . :c)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:23
+   |
+ 3 |  (define-class (C :a :b (:a :b . :c)))
+   |                         ^^^^^^^^^^^^ unexpected dotted list
+
+================================================================================
+116 Malformed functional dependency: missing type variables
+================================================================================
+
+(package test-class)
+
+(define-class (C :a :b (:a :b ->)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:33
+   |
+ 3 |  (define-class (C :a :b (:a :b ->)))
+   |                                  ^ expected one or more type variables
+
+================================================================================
+117 Malformed functional dependency: missing type variables (2)
+================================================================================
+
+(package test-package)
+
+(define-class (C :a :b (-> :b :c)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:23
+   |
+ 3 |  (define-class (C :a :b (-> :b :c)))
```

Comment:
This too, it should probably refer to the arrow, similarly to the => errors above .

(Once again, probably a separate PR)